[TopBar.Search] Fix search dismissal overlay (#2440)
Shopify/polaris-react: UNRELEASED.md

### Enhancements

> - Updated `Popover` to focus the correct element when closed ([#2255](https://github.com/Shopify/polaris-react/pull/2255))

Reviewer: You've got some stuff here that shouldn't be. These lines were probably included in a release but and got mixed up in a rebase or something.

Author: This has been fixed now I think 💯

### Enhancements

- Added `searchResultsOverlayVisible` prop to `TopBar` which adds a translucent background to the search dismissal overlay when results are displayed ([#2440](https://github.com/Shopify/polaris-react/pull/2440))

### Bug fixes

> - Fixed `TextField` to no longer render `aria-invalid="false"` ([#2282](https://github.com/Shopify/polaris-react/pull/2282))

Reviewer: ```suggestion
```

> - Fixed `TextField` to only render `min` ,`max` and `step` attributes when explicitly passed ([#2282](https://github.com/Shopify/polaris-react/pull/2282))

Reviewer: ```suggestion
```

### Bug fixes

> - Added opaque dismissal overlay to `Search` when results are displayed ([#2440](https://github.com/Shopify/polaris-react/pull/2440))

Reviewer: ```suggestion
- Added `searchResultsOverlayVisible` prop to `TopBar` which adds a translucent background to the search dismissal overlay when results are displayed ([#2440](https://github.com/Shopify/polaris-react/pull/2440))
```
